src/index.coffee

```diff
@@ -117,15 +118,33 @@ progenyConstructor = (mode, settings = {}) ->
 				if path in depsList
 					callback()
 				else
-					depsList.push path if potentialDeps
-					fs[mode].readFile path, encoding: 'utf8', (err, source) ->
-						return callback() if err
-						depsList.push path unless potentialDeps or path in depsList
-						parseDeps path, source, depsList, callback
+					if glob.hasMagic path
```

Comment:
Glob checking should also be behind a new option and enabled for specific formats that support it